### PR TITLE
LPS-146461 baseline

### PR DESCRIPTION
**Pull request by ealonso**
Hey @liferay-forms,

I'm sending it to you to review the first three commits, since @bakayattila changed the StructureNameComparator on dynamic data mapping modules to allow ordering structure names by locale, also he added some tests to check the new behavior.

I've already reviewed the code from Echo modules.

Could you please take a look at it and forward it to Brian if everything looks good to you?

For more information see: https://issues.liferay.com/browse/PTR-2856 and https://issues.liferay.com/browse/LPS-146461

Thanks!!

### CODE REVIEW

**Comment by liferay-forms** [comment]
ci:test:sf

**Comment by liferay-forms** [comment]
Hi, you don’t need to run the CI commands anymore, I’ll do it for you and let you know if any issue happens.

**Comment by liferay-forms** [comment]
ci:test:relevant

**Comment by liferay-forms** [comment]
Hey @ealonso, can you please check if these failures are related to your changes? Thanks

**Comment by ealonso** [comment]
ci:reevaluate:1327401_2441

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/util/test/StructureNameComparatorTest.java [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+
+		_ddmStructure4 = DDMStructureTestUtil.addStructure(
+			_group.getGroupId(), JournalArticle.class.getName(), defaultLocale);
+
+		_ddmStructure4.setName("default name D", defaultLocale);
+	}
+
+	@Test
+	public void testComparatorWithDefaultLocale() {
+		List<DDMStructure> structureList = new ArrayList<>();
```
Comment (carolmariaabb) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/util/test/StructureNameComparatorTest.java:85]: @bakayattila @ealonso please rename it to `ddmStructures`?

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/util/test/StructureNameComparatorTest.java [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+
+		Assert.assertEquals(_ddmStructure1, structureList.get(0));
+		Assert.assertEquals(_ddmStructure2, structureList.get(1));
+		Assert.assertEquals(_ddmStructure3, structureList.get(2));
+		Assert.assertEquals(_ddmStructure4, structureList.get(3));
+	}
+
+	@Test
+	public void testComparatorWithLocale() {
+		List<DDMStructure> structureList = new ArrayList<>();
```
Comment (carolmariaabb) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/util/test/StructureNameComparatorTest.java:104]: @bakayattila @ealonso please rename it to `ddmStructures`?

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/util/test/StructureNameComparatorTest.java [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new LiferayIntegrationTestRule();
+
+	@Before
+	public void setUp() throws Exception {
+		_group = GroupTestUtil.addGroup();
+
+		Locale defaultLocale = LocaleUtil.getSiteDefault();
+		_otherLocale = LocaleUtil.SPAIN;
```
Comment (carolmariaabb) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/util/test/StructureNameComparatorTest.java:57]: @bakayattila @ealonso can we change the declaration of this variable, as follows, so we can remove this line, and rename it to `esLocale`?
```
private final Locale _esLocale = LocaleUtil.SPAIN;
```

**Comment by ealonso** [comment]
Hey @carolmariaabb,

I've added two new commits with the requested changes.

Regards,
Eudaldo.

cc: @bakayattila

**Review by carolmariaabb — APPROVED** [review]

**Comment by liferay-forms** [comment]
ci:test:relevant

**Comment by liferay-forms** [comment]
Hey @ealonso, can you please check if these failures are related to your changes? Thanks

**Comment by carolmariaabb** [comment]
ci:forward

**Pull request closed without merging**